LoopKit/PumpManager.swift

```diff
@@ -118,9 +147,102 @@ public protocol PumpManager: DeviceManager {
     ///   - result: A DoseEntry or an error describing why the command failed
     func enactTempBasal(unitsPerHour: Double, for duration: TimeInterval, completion: @escaping (_ result: PumpManagerResult<DoseEntry>) -> Void)
 
-    func updateBLEHeartbeatPreference()
-    
+    /// Send a command to the pump to suspend delivery
+    ///
+    /// - Parameters:
+    ///   - completion: A closure called after the command is complete
+    ///   - error: An error describing why the command failed
     func suspendDelivery(completion: @escaping (_ error: Error?) -> Void)
-    
+
+    /// Send a command to the pump to resume delivery
+    ///
+    /// - Parameters:
+    ///   - completion: A closure called after the command is complete
+    ///   - error: An error describing why the command failed
     func resumeDelivery(completion: @escaping (_ error: Error?) -> Void)
 }
+
+
+public extension PumpManager {
+    func roundToSupportedBasalRate(unitsPerHour: Double) -> Double {
+        return supportedBasalRates.filter({$0 <= unitsPerHour}).max() ?? 0
+    }
+
+    func roundToSupportedBolusVolume(units: Double) -> Double {
+        return supportedBolusVolumes.filter({$0 <= units}).max() ?? 0
```

Comment:
Since `supportedBasalRates` and `supportedBolusVolumes` should never be empty, you can force unwrap the calls to `max()` here.